main/lsp/requests/rename.cc

```diff
@@ -114,15 +268,37 @@ unique_ptr<ResponseMessage> RenameTask::runRequest(LSPTypecheckerDelegate &typec
             auto resp = move(queryResponses[0]);
             // Only supports rename requests from constants and class definitions.
             if (auto constResp = resp->isConstant()) {
+                // Sanity check the text.
+                if (islower(params->newName[0])) {
+                    response->error = make_unique<ResponseError>((int)LSPErrorCodes::InvalidRequest,
+                                                                 "Constant names must begin with an uppercase letter.");
+                    return response;
+                }
                 if (isValidRenameLocation(constResp->symbol, gs, response)) {
                     response->result = getRenameEdits(typechecker, constResp->symbol, params->newName);
                 }
             } else if (auto defResp = resp->isDefinition()) {
-                if (defResp->symbol.data(gs)->isClassOrModule()) {
+                if (defResp->symbol.data(gs)->isClassOrModule() && islower(params->newName[0])) {
+                    response->error =
+                        make_unique<ResponseError>((int)LSPErrorCodes::InvalidRequest,
+                                                   "Class and Module names must begin with an uppercase letter.");
+                    return response;
+                }
+
+                if (defResp->symbol.data(gs)->isMethod() && isupper(params->newName[0])) {
```

Comment:
This is missing in `isSend()`.

Maybe we should move new name sanity checks to the renamer or something.

Reply:
You can test this error in the test suite, btw.